### ISSUE TITLE
Community template: Credit card test data generator

### ISSUE THREAD

**Issue by PriteshKiri** [ToolJet/ToolJet#11007]
**Description:**
This Credit Card Test Data Generator template allows developers to quickly create realistic, randomized credit card information for testing purposes, enhancing the efficiency of payment system development and quality assurance processes.

**Template feature list:**
Please check [here](https://docs.google.com/document/d/1RJ_yA36urnge6zCMNkt98qj3oEliSK0WFY6PWS-TzYQ/edit?usp=sharing)

**Template guidelines:**
Please check [here](https://docs.google.com/document/d/1lSXOpU4zXK8lsnF6Afbz1Qle_wW10ytLDV75II7K6yM/edit?usp=sharing)

**Template deadline:**
Upon assignment of a template development task, you must complete the development process within **7 working days** and submit the Pull Request (PR) for testing. If a delay is anticipated, promptly notify the ToolJet team via the [#contributors](https://tooljet.slack.com/archives/C05SSQF138F) Slack channel.
Failure to complete the development within **7 working days** or provide a response regarding any potential delays may result in the reassignment of the PR to other available community developers.

**Comment by sikehish** [comment]
I'd like to take this up!

**Comment by PriteshKiri** [comment]
Hey @sikehish , We greatly appreciate your interest. 

However, we would like to wait and see if there are any first-time contributors who are also interested in this opportunity. 

We hope you understand our decision. Thanks!

**Comment by sikehish** [comment]
> Hey @sikehish , We greatly appreciate your interest.
> 
> However, we would like to wait and see if there are any first-time contributors who are also interested in this opportunity.
> 
> We hope you understand our decision. Thanks!

Sure. Let me know if I can contribute to any other issues :)

**Comment by skdas20** [comment]
I am first time contributor I wanna contribute to this

**Comment by AbhinavTheDev** [comment]
I also want to work on this

**Comment by PriteshKiri** [comment]
@skdas20 assigned. Happy Hacktoberfest! 🎃  

Please make sure that,
 - You sign the CLA after submitting your PR. 
 - You complete this task in 7 days from the day of the assignment.   
 
Thank you for showing interest in contributing to ToolJet! Feel free to reach out to us in our [Slack community](https://join.slack.com/t/tooljet/shared_invite/zt-2rk4w42t0-ZV_KJcWU9VL1BBEjnSHLCA) if you have any questions.

**Comment by PriteshKiri** [comment]
Hey @AbhinavTheDev , this issue has been assigned.  Please check out other open issues under the `community contribution`.

**Comment by mayurathavale18** [comment]
I am up to take this issue, please assign this to me

**Comment by sahilborse** [comment]
@PriteshKiri assign this issue to me I am first timer #11007

**Comment by DeepanshuYadav-code** [comment]
hey @PriteshKiri can you assign this issue #11007 to me under hacktoberfest.

I am a first time contributor in this organization. If the previous the deadline to submit a pr for the previous asignee has been completed then can this issue be assigned to me

**Comment by PriteshKiri** [comment]
Hey @mayurathavale18 @sahilborse @DeepanshuYadav-code , this issue has been assigned.  Please check out other open issues under the community contribution label.

**Comment by PriteshKiri** [comment]
Hey @skdas20  ,

Thank you so much for your contribution! 🙌

Please keep an eye on your PR, as we may add some review comments if needed before merging it.

**Comment by GrowwithNaveen** [comment]
Hi @skdas20,

There are a few issues with the submitted PR related to the feature list. Could you please review these and resubmit?
1. The design doesn’t match the wireframe provided in the feature list. Please ensure the font size and component placement are accurate.
2. Make sure the default values are set correctly for each component.
3. The heading is missing "Output in JSON."
4. The output box is not wired with any queries.
![CleanShot 2024-10-30 at 18 34 57@2x](https://github.com/user-attachments/assets/7506ed14-ea9f-4b6c-bb35-0decba4469a8)



Please make these changes and resubmit the PR for testing.

Thanks!

**Comment by PriteshKiri** [comment]
Hey @skdas20,

Can you please resubmit the PR with the above changes?

**Comment by skdas20** [comment]
Yeah I can submit again
The output box is wired with a .js script I don't know why it's not showing
in your SS

**Comment by PriteshKiri** [comment]
Cool! Please let us know once you share the updated PR.

If you have any queries related to the template feel free to drop it in the contributor's channel of our [Slack community](https://join.slack.com/t/tooljet/shared_invite/zt-2rk4w42t0-ZV_KJcWU9VL1BBEjnSHLCA).

**Comment by PriteshKiri** [comment]
Hey @skdas20,
Any updates on the issue?

**Comment by RudraPratik30** [comment]
Unassigned due to lack of progress.
This is now open for contributions

**Comment by sikehish** [comment]
I'd like to take this up!

**Comment by sikehish** [comment]
> Hey @sikehish, Assigned!
> 
> Thank you for showing interest in contributing to ToolJet! Feel free to reach out to us in our [Slack community](https://join.slack.com/t/tooljet/shared_invite/zt-2rk4w42t0-ZV_KJcWU9VL1BBEjnSHLCA) if you have any questions.
> 
> Happy Coding! 🎉

Thank you I'm working on this.

**Comment by PriteshKiri** [comment]
Hey @sikehish  ,
Thank you so much for your contribution! 🙌
Please keep an eye on your PR, as we may add some review comments if needed before merging it.

**Comment by GrowwithNaveen** [comment]
Hi @sikehish,
The application looks great! Fantastic work—keep it up. We’ll take over this deployment from here and keep you updated on any progress.
Thanks again, and happy coding! 🚀

**Comment by tysoncung** [comment]
Good catch on improving test coverage! I have experience with testing and would be interested in helping. Which modules or functions need tests most urgently?

**Comment by Uttkarshchambiyal** [comment]
Hi, I'm a CSE student with hands-on experience in JavaScript and React (MERN stack). I'd love to work on this community template!

I can build the **Credit Card Test Data Generator** template that generates realistic, randomized card data (card number, CVV, expiry date, cardholder name). I'm familiar with ToolJet's template structure and will follow the template guidelines and deadlines mentioned above.

Please assign this issue to me. Happy to get started right away!

**Comment by mraniket404** [comment]
Hi, I'm interested in working on this issue for GSoC 2026.

I have experience with React, JavaScript, and MERN stack.

I need this issue for my GSoC proposal which is due tomorrow (March 31).
I will submit a Draft PR today itself and complete the full implementation within 7 days.

Please assign this issue to me. Thank you!

**Comment by jakepaul8922-dotcom** [comment]
.

**Comment by mraniket404** [comment]
"Hi @RudraPratik30, since this issue is now unassigned, could you please assign it to me? My GSoC proposal deadline is today. I already have experience with React/JS and will submit a draft PR today. Thank you!"

**Comment by Gaurav8957999847** [comment]
Hi @PriteshKiri,

I would like to work on the Credit Card Test Data Generator community template.

I have experience in Node.js backend development and strong problem-solving skills. I can implement the Luhn algorithm and generate realistic, randomized card data as per the template requirements.

I have reviewed the feature list and guidelines and can complete the task within the 7-working-day timeline.

This will be my first open-source contribution, and I’m ready to start immediately.
Please assign this issue to me.

**Comment by Jim1874** [comment]
Hi @PriteshKiri, I would like to work on this as a first-time contributor. I have reviewed the issue details and can complete it within the 7 working day window. Please assign it to me, thank you!

**Comment by RudraPratik30** [comment]
Hey everyone, thanks for your interest in contributing to this issue, but the team has decided to take it off the roadmap, so we are closing this issue.
